grassroots-backend/src/contacts/Contacts.service.ts

```diff
@@ -6,35 +6,36 @@ import {
   PaginatedContactOutDTO,
   PaginatedContactSearchInDTO,
 } from "../grassroots-shared/Contact.entity.dto.js";
-import { EntityManager, FilterQuery } from "@mikro-orm/core";
+import { EntityRepository, FilterQuery } from "@mikro-orm/core";
 import { LikeOrUndefined } from "../util/LikeOrUndefined";
+import { EntityManagerProvider } from "../orm/EntityManager.provider.js";
 
 @Injectable()
 export class ContactsService {
-  constructor(private readonly entityManager: EntityManager) {}
+  repo: EntityRepository<ContactEntityOutDTO>;
+  constructor(private readonly entityManagerProvider: EntityManagerProvider) {
+    this.repo =
+      entityManagerProvider.entityManager.getRepository<ContactEntityOutDTO>(
+        ContactEntityOutDTO,
```

Comment:
Continuing from https://github.com/gpo/grassroots/pull/77#discussion_r2134815188

In a fully featured dependency injection framework, I would expect to be able to inject the `repo: EntityRepository<ContactEntityOutDTO>` here.

Reply:
Without care, the trouble is that if we inject the repo, the repo gets the global EntityManager instead of the one we control for testing. We could probably do it with a custom @InjectRepository decorator?